After upgrade to Android 8.1, Syncthing stays disabled on named wifi networks

See title.  Was working fine.  Upgraded Android, now not working.  "Syncthing is disabled" message.  If allow it to connect to any wifi network, it starts working again.  Only when I try to "Restrict to certain wifi networks" does it disable itself again.  If I flip that back and forth, then Syncthing will continue to renable and disable itself.

I see some other "Syncthing is disabled" bug reports here, but I'm not sure whether this is the same issue.

    App Version: 0.10.1
    Syncthing Version: not listed in app?
    Android Version: Android 8.1.0, patch level Dec. 5

Nutomic and capi, thank you for your help.  Here are the logs and the steps I go through:

1) Reboot phone.  Wait a few minutes for everything to settle.
2) There is now a notification on phone that says, "Syncthing is disabled"
3) Select the Syncthing app. Says, "Syncthing is disabled, Do you want to change your preferences?"  I click on "CHANGE SETTINGS"
4) I get the logs at this point.  In all the following logs, I've scrubbed anything that looked personally identifiable with "{concealed blah blah blah}".  The logs are:

```
--------- beginning of main
```

and

```
--------- beginning of main
12-15 09:24:49.376 I/zygote64(11329): Do partial code cache collection, code=28KB, data=19KB
12-15 09:24:49.376 I/zygote64(11329): After code cache collection, code=26KB, data=18KB
12-15 09:24:49.377 I/zygote64(11329): Increasing code cache capacity to 128KB
12-15 09:24:49.405 I/DeviceStateHolder(11329): Listening for network state changes
12-15 09:24:49.410 I/DeviceStateHolder(11329): Stopped listening to battery state changes
12-15 09:24:49.410 I/DeviceStateHolder(11329): Listening to power saving changes
12-15 09:24:49.414 I/SyncthingService(11329): Stopping syncthing according to current state and preferences
12-15 09:24:49.414 I/SyncthingService(11329): Shutting down background service
12-15 09:24:49.418 W/Notification(11329): Use of stream types is deprecated for operations other than volume control
12-15 09:24:49.418 W/Notification(11329): See the documentation of setSound() for what to use instead with android.media.AudioAttributes to qualify your playback use case
12-15 09:24:49.427 W/Notification(11329): Use of stream types is deprecated for operations other than volume control
12-15 09:24:49.427 W/Notification(11329): See the documentation of setSound() for what to use instead with android.media.AudioAttributes to qualify your playback use case
12-15 09:24:49.429 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 09:24:49.433 W/DeviceStateHolder(11329): SSID [<unknown ssid>] unknown or not whitelisted, disallowing sync.
12-15 09:24:49.433 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 09:24:49.434 W/DeviceStateHolder(11329): SSID [<unknown ssid>] unknown or not whitelisted, disallowing sync.
12-15 09:24:49.437 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 09:24:49.439 W/DeviceStateHolder(11329): SSID [<unknown ssid>] unknown or not whitelisted, disallowing sync.
12-15 09:27:36.189 I/WebViewFactory(11329): Loading com.android.chrome version 62.0.3202.84 (code 320208452)
12-15 09:27:36.251 I/cr_LibraryLoader(11329): Time to load native libraries: 4 ms (timestamps 6251-6255)
12-15 09:27:36.282 I/chromium(11329): [INFO:library_loader_hooks.cc(46)] Chromium logging enabled: level = 0, default verbosity = 0
12-15 09:27:36.283 I/cr_LibraryLoader(11329): Expected native library version number "62.0.3202.84", actual native library version number "62.0.3202.84"
12-15 09:27:36.296 W/cr_ChildProcLH(11329): Create a new ChildConnectionAllocator with package name = com.android.chrome, sandboxed = true
12-15 09:27:36.303 I/cr_BrowserStartup(11329): Initializing chromium process, singleProcess=false
12-15 09:27:36.324 W/yncthingandroid(11329): type=1400 audit(0.0:164): avc: denied { read } for name="vmstat" dev="proc" ino=4026548020 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:proc:s0 tclass=file permissive=0
12-15 09:27:36.330 I/Adreno  (11329): QUALCOMM build                   : 8e59954, I0be83d0d26
12-15 09:27:36.330 I/Adreno  (11329): Build Date                       : 09/22/17
12-15 09:27:36.330 I/Adreno  (11329): OpenGL ES Shader Compiler Version: EV031.21.02.00
12-15 09:27:36.330 I/Adreno  (11329): Local Branch                     : O17A
12-15 09:27:36.330 I/Adreno  (11329): Remote Branch                    :
12-15 09:27:36.330 I/Adreno  (11329): Remote Branch                    :
12-15 09:27:36.330 I/Adreno  (11329): Reconstruct Branch               :
12-15 09:27:36.337 I/Adreno  (11329): PFP: 0x00000000, ME: 0x00000000
12-15 09:27:36.346 I/zygote64(11329): android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 0
12-15 09:27:36.615 I/ConfigXml(11329): Loaded Syncthing config file
12-15 09:27:36.714 I/OpenGLRenderer(11329): Initialized EGL, version 1.4
12-15 09:27:36.857 W/cr_CrashFileManager(11329): /data/user/0/com.nutomic.syncthingandroid/cache/WebView/Crash Reports does not exist or is not a directory
12-15 09:29:01.016 I/zygote64(11329): Do partial code cache collection, code=58KB, data=43KB
12-15 09:29:01.018 I/zygote64(11329): After code cache collection, code=58KB, data=43KB
12-15 09:29:01.018 I/zygote64(11329): Increasing code cache capacity to 256KB
12-15 09:29:03.534 I/zygote64(11329): Deoptimizing void android.support.v7.widget.AppCompatTextHelper.updateTypefaceAndStyle(android.content.Context, android.support.v7.widget.TintTypedArray) due to JIT inline cache
12-15 09:29:45.220 I/zygote64(11329): Deoptimizing android.graphics.Typeface android.support.v7.widget.TintTypedArray.getFont(int, int, android.support.v4.content.res.ResourcesCompat$FontCallback) due to JIT inline cache

```

5) Now go to "Run Conditions" (there, I already have "Run only on wifi" checked), click "Restrict to certain wifi networks" (there, I already have all networks unchecked EXCEPT my home network, "Sallys5.0").  I now uncheck that Sallys5.0, which means all networks are unchecked, which means it should now run on any wifi network.  Click "OK".

6) Now the notification says, "Syncthing is running".  Seems to be working fine now.  Here are the logs at that point:

```
--------- beginning of main
12-15 09:41:04.043 I/SyncthingNativeCode(11329): [FHRWO] INFO: syncthing v0.14.41 "Dysprosium Dragonfly" (go1.9.2 android-arm64) felix@p50 2017-11-23 07:29:56 UTC [noupgrade]
12-15 09:41:04.047 I/SyncthingNativeCode(11329): [FHRWO] INFO: My ID: {concealed blah blah blah}
12-15 09:41:04.047 I/SyncthingNativeCode(11329): [FHRWO] INFO: Disabling weak hash
12-15 09:41:04.143 I/SyncthingNativeCode(11329): [FHRWO] INFO: Ready to synchronize "6p-Desktop-Laptop" ({concealed blah blah blah}) (readwrite)
12-15 09:41:04.144 I/SyncthingNativeCode(11329): [FHRWO] INFO: Send rate is unlimited, receive rate is unlimited
12-15 09:41:04.144 I/SyncthingNativeCode(11329): [FHRWO] INFO: Rate limits do not apply to LAN connections
12-15 09:41:04.149 I/SyncthingNativeCode(11329): [FHRWO] INFO: KCP listener ([::]:22020) starting
12-15 09:41:04.151 I/SyncthingNativeCode(11329): [FHRWO] INFO: TCP listener ([::]:22000) starting
12-15 09:41:04.166 I/SyncthingNativeCode(11329): [FHRWO] INFO: Completed initial scan of readwrite folder "6p-Desktop-Laptop" ({concealed blah blah blah})
12-15 09:41:04.230 I/SyncthingNativeCode(11329): [FHRWO] INFO: Device {concealed blah blah blah} is "Nexus-6P" at [dynamic]
12-15 09:41:04.231 I/SyncthingNativeCode(11329): [FHRWO] INFO: Device {concealed blah blah blah} is "Desktop-Cinnamon" at [dynamic]
12-15 09:41:04.232 I/SyncthingNativeCode(11329): [FHRWO] INFO: No automatic upgrades; STNOUPGRADE environment variable defined.
12-15 09:41:04.236 I/SyncthingNativeCode(11329): [FHRWO] INFO: GUI and API listening on 127.0.0.1:8384
12-15 09:41:04.237 I/SyncthingNativeCode(11329): [FHRWO] INFO: Access the GUI via the following URL: https://127.0.0.1:8384/
12-15 09:41:04.910 W/SyncthingNativeCode(11329): ionice: exec 14079: Permission denied
12-15 09:41:05.851 I/SyncthingNativeCode(11329): [FHRWO] INFO: Established secure connection to {concealed blah blah blah} at 192.168.1.225:39596-192.168.1.202:22000 (tcp-client) (TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305)
12-15 09:41:05.853 I/SyncthingNativeCode(11329): [FHRWO] INFO: Device {concealed blah blah blah} client is "syncthing v0.14.41" named "Desktop-Cinnamon"
12-15 09:41:43.063 I/SyncthingNativeCode(11329): [FHRWO] INFO: kcp://0.0.0.0:22020 detected NAT type: Port restricted NAT
12-15 09:41:43.067 I/SyncthingNativeCode(11329): [FHRWO] INFO: kcp://0.0.0.0:22020 resolved external address kcp://{concealed blah blah blah}:1024 (via stun.voipbuster.com:3478)
```

and

```
--------- beginning of main
12-15 09:24:49.429 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 09:24:49.433 W/DeviceStateHolder(11329): SSID [<unknown ssid>] unknown or not whitelisted, disallowing sync.
12-15 09:24:49.433 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 09:24:49.434 W/DeviceStateHolder(11329): SSID [<unknown ssid>] unknown or not whitelisted, disallowing sync.
12-15 09:24:49.437 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 09:24:49.439 W/DeviceStateHolder(11329): SSID [<unknown ssid>] unknown or not whitelisted, disallowing sync.
12-15 09:27:36.189 I/WebViewFactory(11329): Loading com.android.chrome version 62.0.3202.84 (code 320208452)
12-15 09:27:36.251 I/cr_LibraryLoader(11329): Time to load native libraries: 4 ms (timestamps 6251-6255)
12-15 09:27:36.282 I/chromium(11329): [INFO:library_loader_hooks.cc(46)] Chromium logging enabled: level = 0, default verbosity = 0
12-15 09:27:36.283 I/cr_LibraryLoader(11329): Expected native library version number "62.0.3202.84", actual native library version number "62.0.3202.84"
12-15 09:27:36.296 W/cr_ChildProcLH(11329): Create a new ChildConnectionAllocator with package name = com.android.chrome, sandboxed = true
12-15 09:27:36.303 I/cr_BrowserStartup(11329): Initializing chromium process, singleProcess=false
12-15 09:27:36.324 W/yncthingandroid(11329): type=1400 audit(0.0:164): avc: denied { read } for name="vmstat" dev="proc" ino=4026548020 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:proc:s0 tclass=file permissive=0
12-15 09:27:36.330 I/Adreno  (11329): QUALCOMM build                   : 8e59954, I0be83d0d26
12-15 09:27:36.330 I/Adreno  (11329): Build Date                       : 09/22/17
12-15 09:27:36.330 I/Adreno  (11329): OpenGL ES Shader Compiler Version: EV031.21.02.00
12-15 09:27:36.330 I/Adreno  (11329): Local Branch                     : O17A
12-15 09:27:36.330 I/Adreno  (11329): Remote Branch                    :
12-15 09:27:36.330 I/Adreno  (11329): Remote Branch                    :
12-15 09:27:36.330 I/Adreno  (11329): Reconstruct Branch               :
12-15 09:27:36.337 I/Adreno  (11329): PFP: 0x00000000, ME: 0x00000000
12-15 09:27:36.346 I/zygote64(11329): android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 0
12-15 09:27:36.615 I/ConfigXml(11329): Loaded Syncthing config file
12-15 09:27:36.714 I/OpenGLRenderer(11329): Initialized EGL, version 1.4
12-15 09:27:36.857 W/cr_CrashFileManager(11329): /data/user/0/com.nutomic.syncthingandroid/cache/WebView/Crash Reports does not exist or is not a directory
12-15 09:29:01.016 I/zygote64(11329): Do partial code cache collection, code=58KB, data=43KB
12-15 09:29:01.018 I/zygote64(11329): After code cache collection, code=58KB, data=43KB
12-15 09:29:01.018 I/zygote64(11329): Increasing code cache capacity to 256KB
12-15 09:29:03.534 I/zygote64(11329): Deoptimizing void android.support.v7.widget.AppCompatTextHelper.updateTypefaceAndStyle(android.content.Context, android.support.v7.widget.TintTypedArray) due to JIT inline cache
12-15 09:29:45.220 I/zygote64(11329): Deoptimizing android.graphics.Typeface android.support.v7.widget.TintTypedArray.getFont(int, int, android.support.v4.content.res.ResourcesCompat$FontCallback) due to JIT inline cache
12-15 09:29:55.301 I/zygote64(11329): Do full code cache collection, code=122KB, data=85KB
12-15 09:29:55.302 I/zygote64(11329): After code cache collection, code=79KB, data=48KB
12-15 09:30:02.583 I/zygote64(11329): Explicit concurrent copying GC freed 27175(1910KB) AllocSpace objects, 7(136KB) LOS objects, 49% free, 4MB/8MB, paused 65us total 46.299ms
12-15 09:30:02.632 I/zygote64(11329): Explicit concurrent copying GC freed 7135(324KB) AllocSpace objects, 0(0B) LOS objects, 50% free, 4MB/8MB, paused 42us total 33.850ms
12-15 09:38:10.347 W/InputEventReceiver(11329): Attempted to finish an input event but the input event receiver has already been disposed.
12-15 09:41:03.590 I/DeviceStateHolder(11329): Listening for network state changes
12-15 09:41:03.595 I/DeviceStateHolder(11329): Stopped listening to battery state changes
12-15 09:41:03.595 I/DeviceStateHolder(11329): Listening to power saving changes
12-15 09:41:03.598 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 09:41:03.601 I/SyncthingService(11329): Shutting down background service
12-15 09:41:03.604 W/Notification(11329): Use of stream types is deprecated for operations other than volume control
12-15 09:41:03.604 W/Notification(11329): See the documentation of setSound() for what to use instead with android.media.AudioAttributes to qualify your playback use case
12-15 09:41:03.607 I/SyncthingService(11329): Starting syncthing according to current state and preferences
12-15 09:41:03.610 W/Notification(11329): Use of stream types is deprecated for operations other than volume control
12-15 09:41:03.610 W/Notification(11329): See the documentation of setSound() for what to use instead with android.media.AudioAttributes to qualify your playback use case
12-15 09:41:03.613 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 09:41:03.617 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 09:41:03.634 I/ConfigXml(11329): Loaded Syncthing config file
12-15 09:41:03.634 I/ConfigXml(11329): Checking for needed config updates
12-15 09:41:03.700 I/SyncthingService(11329): Web GUI will be available at https://127.0.0.1:8384
12-15 09:41:03.700 I/PollWebGuiAvailableTask(11329): Starting to poll for web gui availability
12-15 09:41:03.804 W/chmod   (14076): type=1400 audit(0.0:166): avc: denied { setattr } for name="libsyncthing.so" dev="dm-2" ino=7340228 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:apk_data_file:s0 tclass=file permissive=0
12-15 09:41:04.043 I/SyncthingNativeCode(11329): [FHRWO] INFO: syncthing v0.14.41 "Dysprosium Dragonfly" (go1.9.2 android-arm64) felix@p50 2017-11-23 07:29:56 UTC [noupgrade]
12-15 09:41:04.047 I/SyncthingNativeCode(11329): [FHRWO] INFO: My ID: {concealed blah blah blah}
12-15 09:41:04.047 I/SyncthingNativeCode(11329): [FHRWO] INFO: Disabling weak hash
12-15 09:41:04.143 I/SyncthingNativeCode(11329): [FHRWO] INFO: Ready to synchronize "6p-Desktop-Laptop" ({concealed blah blah blah}) (readwrite)
12-15 09:41:04.144 I/SyncthingNativeCode(11329): [FHRWO] INFO: Send rate is unlimited, receive rate is unlimited
12-15 09:41:04.144 I/SyncthingNativeCode(11329): [FHRWO] INFO: Rate limits do not apply to LAN connections
12-15 09:41:04.149 I/SyncthingNativeCode(11329): [FHRWO] INFO: KCP listener ([::]:22020) starting
12-15 09:41:04.151 I/SyncthingNativeCode(11329): [FHRWO] INFO: TCP listener ([::]:22000) starting
12-15 09:41:04.154 W/libsyncthing.so(14095): type=1400 audit(0.0:167): avc: denied { read } for name="hostname" dev="proc" ino=192018 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:proc:s0 tclass=file permissive=0
12-15 09:41:04.166 I/SyncthingNativeCode(11329): [FHRWO] INFO: Completed initial scan of readwrite folder "6p-Desktop-Laptop" ({concealed blah blah blah})
12-15 09:41:04.230 I/SyncthingNativeCode(11329): [FHRWO] INFO: Device {concealed blah blah blah} is "Nexus-6P" at [dynamic]
12-15 09:41:04.231 I/SyncthingNativeCode(11329): [FHRWO] INFO: Device {concealed blah blah blah} is "Desktop-Cinnamon" at [dynamic]
12-15 09:41:04.232 I/SyncthingNativeCode(11329): [FHRWO] INFO: No automatic upgrades; STNOUPGRADE environment variable defined.
12-15 09:41:04.236 I/SyncthingNativeCode(11329): [FHRWO] INFO: GUI and API listening on 127.0.0.1:8384
12-15 09:41:04.237 I/SyncthingNativeCode(11329): [FHRWO] INFO: Access the GUI via the following URL: https://127.0.0.1:8384/
12-15 09:41:04.380 I/SyncthingService(11329): Web GUI has come online at https://127.0.0.1:8384
12-15 09:41:04.381 W/Notification(11329): Use of stream types is deprecated for operations other than volume control
12-15 09:41:04.381 W/Notification(11329): See the documentation of setSound() for what to use instead with android.media.AudioAttributes to qualify your playback use case
12-15 09:41:04.408 W/cr_media(11329): Requires BLUETOOTH permission
12-15 09:41:04.418 I/cr_LibraryLoader(11329): Using linker: org.chromium.base.library_loader.ModernLinker
12-15 09:41:04.450 I/RestApi (11329): Syncthing version is v0.14.41
12-15 09:41:04.453 W/Notification(11329): Use of stream types is deprecated for operations other than volume control
12-15 09:41:04.453 W/Notification(11329): See the documentation of setSound() for what to use instead with android.media.AudioAttributes to qualify your playback use case
12-15 09:41:04.508 I/FolderObserver(11329): Observer created for (folder {concealed blah blah blah})
12-15 09:41:04.564 I/zygote64(11329): Do partial code cache collection, code=117KB, data=79KB
12-15 09:41:04.564 I/zygote64(11329): After code cache collection, code=117KB, data=79KB
12-15 09:41:04.564 I/zygote64(11329): Increasing code cache capacity to 512KB
12-15 09:41:04.576 W/VideoCapabilities(11329): Unrecognized profile 2130706433 for video/avc
12-15 09:41:04.580 I/VideoCapabilities(11329): Unsupported profile 4 for video/mp4v-es
12-15 09:41:04.672 I/X509Util(11329): Failed to validate the certificate chain, error: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.
12-15 09:41:04.910 W/SyncthingNativeCode(11329): ionice: exec 14079: Permission denied
12-15 09:41:04.912 I/SyncthingRunnableIoNice(11329): ionice performed on libsyncthing.so
12-15 09:41:04.912 E/SyncthingRunnableIoNice(11329): Failed to set ionice 127
12-15 09:41:05.851 I/SyncthingNativeCode(11329): [FHRWO] INFO: Established secure connection to {concealed blah blah blah} at 192.168.1.225:39596-192.168.1.202:22000 (tcp-client) (TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305)
12-15 09:41:05.853 I/SyncthingNativeCode(11329): [FHRWO] INFO: Device {concealed blah blah blah} client is "syncthing v0.14.41" named "Desktop-Cinnamon"
12-15 09:41:06.256 I/chromium(11329): [INFO:CONSOLE(1426)] "loadFormIntoScope", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (1426)
12-15 09:41:06.298 I/chromium(11329): [INFO:CONSOLE(1426)] "loadFormIntoScope", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (1426)
12-15 09:41:06.815 I/chromium(11329): [INFO:CONSOLE(116)] "UIOnline", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (116)
12-15 09:41:06.915 I/chromium(11329): [INFO:CONSOLE(566)] "refreshConfig", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (566)
12-15 09:41:07.069 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:41:07.093 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:41:07.202 I/chromium(11329): [INFO:CONSOLE(681)] "refreshfolderStats", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (681)
12-15 09:41:07.241 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:41:07.291 I/chromium(11329): [INFO:CONSOLE(666)] "refreshDeviceStats", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (666)
12-15 09:41:07.427 I/chromium(11329): [INFO:CONSOLE(504)] "recalcCompletion", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (504)
12-15 09:41:07.440 I/chromium(11329): [INFO:CONSOLE(373)] "refreshFolder", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (373)
12-15 09:41:07.475 I/chromium(11329): [INFO:CONSOLE(504)] "recalcCompletion", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (504)
12-15 09:41:18.218 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:41:18.260 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:41:18.272 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:41:18.284 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:41:19.523 I/zygote64(11329): Deoptimizing java.lang.Object com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(com.google.gson.stream.JsonReader) due to JIT inline cache
12-15 09:41:28.223 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:41:28.244 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:41:28.267 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:41:28.301 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:41:38.224 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:41:38.262 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:41:38.284 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:41:38.299 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:41:43.063 I/SyncthingNativeCode(11329): [FHRWO] INFO: kcp://0.0.0.0:22020 detected NAT type: Port restricted NAT
12-15 09:41:43.067 I/SyncthingNativeCode(11329): [FHRWO] INFO: kcp://0.0.0.0:22020 resolved external address kcp://{concealed blah blah blah} (via stun.voipbuster.com:3478)
12-15 09:41:46.116 W/InputEventReceiver(11329): Attempted to finish an input event but the input event receiver has already been disposed.
12-15 09:41:48.196 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:41:48.228 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:41:48.241 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:41:48.249 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:41:48.401 W/InputEventReceiver(11329): Attempted to finish an input event but the input event receiver has already been disposed.
12-15 09:41:58.191 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:41:58.220 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:41:58.241 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:41:58.270 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:42:01.218 I/chromium(11329): [INFO:CONSOLE(681)] "refreshfolderStats", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (681)
12-15 09:42:06.918 I/chromium(11329): [INFO:CONSOLE(696)] "refreshGlobalChanges", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (696)
12-15 09:42:07.244 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:42:07.262 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:42:07.302 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:42:07.327 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
```

7) I now go back to "Restrict certain wifi networks" and checkmark my home network again, and say "OK".  Notifcation changes back to "Syncthing is disabled".  Logs now:

```
--------- beginning of main
12-15 09:41:43.063 I/SyncthingNativeCode(11329): [FHRWO] INFO: kcp://0.0.0.0:22020 detected NAT type: Port restricted NAT
12-15 09:41:43.067 I/SyncthingNativeCode(11329): [FHRWO] INFO: kcp://0.0.0.0:22020 resolved external address kcp://--------- beginning of main
12-15 09:41:43.063 I/SyncthingNativeCode(11329): [FHRWO] INFO: kcp://0.0.0.0:22020 detected NAT type: Port restricted NAT
12-15 09:41:43.067 I/SyncthingNativeCode(11329): [FHRWO] INFO: kcp://0.0.0.0:22020 resolved external address kcp://{concealed blah blah blah} (via stun.voipbuster.com:3478)
12-15 10:06:04.057 I/SyncthingNativeCode(11329): [FHRWO] INFO: Exiting (via stun.voipbuster.com:3478)
12-15 10:06:04.057 I/SyncthingNativeCode(11329): [FHRWO] INFO: Exiting
```

and

```
--------- beginning of main
12-15 09:44:45.085 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:44:45.117 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:44:45.131 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:44:45.154 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:44:45.648 I/chromium(11329): [INFO:CONSOLE(681)] "refreshfolderStats", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (681)
12-15 09:45:07.110 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:45:07.147 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:45:07.164 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:45:07.187 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:47:23.313 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:47:23.331 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:47:23.361 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:47:23.393 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:47:49.460 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:47:51.545 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:47:51.558 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:47:51.565 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:50:22.592 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:50:22.620 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:50:22.637 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:50:22.650 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:51:35.785 I/chromium(11329): [INFO:CONSOLE(681)] "refreshfolderStats", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (681)
12-15 09:51:39.094 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:51:39.124 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:51:39.131 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:51:39.141 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:51:57.646 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:51:57.672 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:51:57.680 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:51:57.689 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:52:14.803 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:52:14.841 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:52:14.859 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:52:14.875 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:52:38.012 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:52:38.024 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:52:38.031 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:52:38.042 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:52:47.846 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:52:47.887 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:52:47.904 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:52:47.913 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:52:57.834 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:52:57.861 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:52:57.879 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:52:57.894 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:52:57.960 I/chromium(11329): [INFO:CONSOLE(681)] "refreshfolderStats", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (681)
12-15 09:53:07.826 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:53:07.854 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:53:07.872 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:53:07.888 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:53:17.832 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:53:17.848 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:53:17.878 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:53:17.894 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:53:27.845 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:53:27.884 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:53:27.902 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:53:27.916 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:53:37.840 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:53:37.864 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:53:37.879 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:53:37.896 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:53:47.841 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:53:47.870 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:53:47.888 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:53:47.901 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:53:57.835 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:53:57.866 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:53:57.879 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:53:57.888 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:54:06.933 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:54:06.960 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:54:06.974 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:54:06.988 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:54:07.025 I/chromium(11329): [INFO:CONSOLE(681)] "refreshfolderStats", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (681)
12-15 09:54:17.840 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:54:17.873 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:54:17.883 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:54:17.892 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:54:27.854 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:54:27.889 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:54:27.904 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:54:27.921 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:54:37.807 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:54:37.819 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:54:37.826 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:54:37.832 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:54:47.811 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:54:47.827 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:54:47.834 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:54:47.848 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:55:26.257 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:55:26.286 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:55:26.302 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:55:26.324 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:56:52.787 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:56:52.814 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:56:52.823 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:56:52.841 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:56:53.670 I/chromium(11329): [INFO:CONSOLE(681)] "refreshfolderStats", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (681)
12-15 09:58:32.198 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:58:32.226 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:58:32.250 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:58:32.260 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 09:59:10.679 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 09:59:10.699 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 09:59:10.710 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 09:59:10.723 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:00:34.450 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:00:34.499 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:00:34.522 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:00:34.530 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:00:43.864 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:00:43.898 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:00:43.912 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:00:43.922 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:01:31.906 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:01:31.923 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:01:31.931 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:01:31.943 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:02:32.602 I/chromium(11329): [INFO:CONSOLE(681)] "refreshfolderStats", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (681)
12-15 10:02:36.290 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:02:36.320 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:02:36.338 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:02:36.347 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:03:02.139 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:03:02.164 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:03:02.175 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:03:02.187 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:04:47.106 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:04:47.133 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:04:47.140 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:04:47.146 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:04:59.423 W/InputEventReceiver(11329): Attempted to finish an input event but the input event receiver has already been disposed.
12-15 10:05:00.049 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:05:00.076 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:05:00.088 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:05:00.096 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:05:10.069 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:05:10.087 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:05:10.101 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:05:10.119 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:05:20.058 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:05:20.088 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:05:20.099 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:05:20.115 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:05:30.063 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:05:30.092 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:05:30.102 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:05:30.118 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:05:30.295 I/chromium(11329): [INFO:CONSOLE(681)] "refreshfolderStats", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (681)
12-15 10:05:40.058 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:05:40.076 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:05:40.092 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:05:40.102 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:05:50.031 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:05:50.053 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:05:50.065 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:05:50.075 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:06:00.054 I/chromium(11329): [INFO:CONSOLE(444)] "refreshSystem", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (444)
12-15 10:06:00.087 I/chromium(11329): [INFO:CONSOLE(460)] "refreshDiscoveryCache", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (460)
12-15 10:06:00.097 I/chromium(11329): [INFO:CONSOLE(559)] "refreshErrors", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (559)
12-15 10:06:00.107 I/chromium(11329): [INFO:CONSOLE(552)] "refreshConnections", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (552)
12-15 10:06:03.956 I/DeviceStateHolder(11329): Listening for network state changes
12-15 10:06:03.959 I/DeviceStateHolder(11329): Stopped listening to battery state changes
12-15 10:06:03.960 I/DeviceStateHolder(11329): Listening to power saving changes
12-15 10:06:03.963 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 10:06:03.964 W/DeviceStateHolder(11329): SSID [<unknown ssid>] unknown or not whitelisted, disallowing sync.
12-15 10:06:03.965 I/SyncthingService(11329): Stopping syncthing according to current state and preferences
12-15 10:06:03.965 I/SyncthingService(11329): Shutting down background service
12-15 10:06:03.966 W/Notification(11329): Use of stream types is deprecated for operations other than volume control
12-15 10:06:03.966 W/Notification(11329): See the documentation of setSound() for what to use instead with android.media.AudioAttributes to qualify your playback use case
12-15 10:06:03.970 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 10:06:03.971 W/DeviceStateHolder(11329): SSID [<unknown ssid>] unknown or not whitelisted, disallowing sync.
12-15 10:06:03.981 I/DeviceStateHolder(11329): State updated, allowed network connection: true, charging: false, power saving: false
12-15 10:06:03.983 W/DeviceStateHolder(11329): SSID [<unknown ssid>] unknown or not whitelisted, disallowing sync.
12-15 10:06:04.057 I/SyncthingNativeCode(11329): [FHRWO] INFO: Exiting
12-15 10:06:04.071 I/SyncthingRunnable(11329): Syncthing exited with code 0
12-15 10:06:04.071 E/Volley  (11329): [355] BasicNetwork.performRequest: Unexpected response code 200 for https://127.0.0.1:8384/rest/events?since=46&limit=0
12-15 10:06:04.072 I/chromium(11329): [INFO:CONSOLE(164)] "UIOffline", source: https://127.0.0.1:8384/syncthing/core/syncthingController.js (164)
12-15 10:06:04.092 W/ApiRequest(11329): Request to https://127.0.0.1:8384/rest/events?since=46&limit=0 failed
12-15 10:06:04.092 W/ApiRequest(11329): com.android.volley.NoConnectionError: java.net.ConnectException: Failed to connect to /127.0.0.1:8384
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:177)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:120)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:87)
12-15 10:06:04.092 W/ApiRequest(11329): Caused by: java.net.ConnectException: Failed to connect to /127.0.0.1:8384
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:143)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.io.RealConnection.connect(RealConnection.java:112)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:461)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:407)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponseCode(HttpURLConnectionImpl.java:538)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getResponseCode(DelegatingHttpsURLConnection.java:105)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getResponseCode(Unknown Source:0)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.volley.toolbox.HurlStack.executeRequest(HurlStack.java:99)
12-15 10:06:04.092 W/ApiRequest(11329): at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:131)
12-15 10:06:04.092 W/ApiRequest(11329): ... 2 more
12-15 10:06:05.074 I/SyncthingRunnableKill(11329): Killed Syncthing process 14079
12-15 10:06:37.484 W/InputEventReceiver(11329): Attempted to finish an input event but the input event receiver has already been disposed.
12-15 10:06:47.167 I/zygote64(11329): Deoptimizing void android.support.v7.view.menu.MenuBuilder.dispatchPresenterUpdate(boolean) due to JIT inline cache
```

It seems to me, that for whatever reason, Android 8.1 doesn't let Syncthing "see" whether I'm connected to my actual home network any more.

Thanks again!

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -14,6 +14,8 @@
     <uses-permission android:name="android.permission.WAKE_LOCK" />
     <uses-permission android:name="android.permission.READ_SYNC_SETTINGS" />
     <uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS"/>
+    <!-- ACCESS_COARSE_LOCATION is required to get WiFi's SSID on 8.1 -->
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
 
     <application
         android:allowBackup="false"

--- a/app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java
@@ -1,5 +1,6 @@
 package com.nutomic.syncthingandroid.activities;
 
+import android.Manifest;
 import android.app.AlertDialog;
 import android.content.Intent;
 import android.content.SharedPreferences;
@@ -13,6 +14,7 @@ import android.preference.ListPreference;
 import android.preference.Preference;
 import android.preference.PreferenceFragment;
 import android.preference.PreferenceScreen;
+import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.util.Log;
 import android.widget.Toast;
@@ -47,6 +49,26 @@ public class SettingsActivity extends SyncthingActivity {
         getFragmentManager().beginTransaction()
                 .replace(android.R.id.content, new SettingsFragment())
                 .commit();
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        // On Android 8.1, ACCESS_COARSE_LOCATION is required, see issue #999
+        if (requestCode == Constants.PERM_REQ_ACCESS_COARSE_LOCATION) {
+            for (int i = 0; i < permissions.length; i++) {
+                if (Manifest.permission.ACCESS_COARSE_LOCATION.equals(permissions[i])) {
+                    if (grantResults[i] == PackageManager.PERMISSION_GRANTED) {
+                        this.startService(new Intent(this, SyncthingService.class)
+                                .setAction(SyncthingService.ACTION_REFRESH_NETWORK_INFO));
+                    } else {
+                        new AlertDialog.Builder(this)
+                                .setTitle(R.string.sync_only_wifi_ssids_location_permission_rejected_dialog_title)
+                                .setMessage(R.string.sync_only_wifi_ssids_location_permission_rejected_dialog_content)
+                                .setPositiveButton(android.R.string.ok, null).show();
+                    }
+                }
+            }
+        }
     }
 
     public static class SettingsFragment extends PreferenceFragment

--- a/app/src/main/java/com/nutomic/syncthingandroid/service/Constants.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/service/Constants.java
@@ -18,6 +18,12 @@ public class Constants {
     public static final String PREF_USE_WAKE_LOCK            = "wakelock_while_binary_running";
 
     /**
+     * On Android 8.1, ACCESS_COARSE_LOCATION is required to access WiFi SSID.
+     * This is the request code used when requesting the permission.
+     */
+    public static final int PERM_REQ_ACCESS_COARSE_LOCATION = 999; // for issue #999
+
+    /**
      * Interval in ms at which the GUI is updated (eg {@link com.nutomic.syncthingandroid.fragments.DrawerFragment}).
      */
     public static final long GUI_UPDATE_INTERVAL = TimeUnit.SECONDS.toMillis(5);

--- a/app/src/main/java/com/nutomic/syncthingandroid/service/DeviceStateHolder.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/service/DeviceStateHolder.java
@@ -183,6 +183,11 @@ public class DeviceStateHolder implements SharedPreferences.OnSharedPreferenceCh
         }
     }
 
+    public void refreshNetworkInfo() {
+        updateWifiSsid();
+        mListener.onDeviceStateChanged();
+    }
+
     /**
      * Determines if Syncthing should currently run.
      */

--- a/app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingService.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingService.java
@@ -48,6 +48,9 @@ public class SyncthingService extends Service {
     public static final String ACTION_RESET =
             "com.nutomic.syncthingandroid.service.SyncthingService.RESET";
 
+    public static final String ACTION_REFRESH_NETWORK_INFO =
+            "com.nutomic.syncthingandroid.service.SyncthingService.REFRESH_NETWORK_INFO";
+
     /**
      * Callback for when the Syncthing web interface becomes first available after service start.
      */
@@ -122,6 +125,8 @@ public class SyncthingService extends Service {
                 new SyncthingRunnable(this, SyncthingRunnable.Command.reset).run();
                 new StartupTask().executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
             });
+        } else if (ACTION_REFRESH_NETWORK_INFO.equals(intent.getAction())) {
+            mDeviceStateHolder.refreshNetworkInfo();
         }
         return START_STICKY;
     }

--- a/app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/views/WifiSsidPreference.java
@@ -1,14 +1,21 @@
 package com.nutomic.syncthingandroid.views;
 
+import android.Manifest;
+import android.app.Activity;
 import android.content.Context;
+import android.content.pm.PackageManager;
 import android.net.wifi.WifiConfiguration;
 import android.net.wifi.WifiManager;
+import android.os.Build;
 import android.os.Bundle;
 import android.preference.MultiSelectListPreference;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
 import android.util.AttributeSet;
 import android.widget.Toast;
 
 import com.nutomic.syncthingandroid.R;
+import com.nutomic.syncthingandroid.service.Constants;
 
 import java.util.Arrays;
 import java.util.HashSet;
@@ -66,6 +73,19 @@ public class WifiSsidPreference extends MultiSelectListPreference {
             super.showDialog(state);
         } else {
             Toast.makeText(getContext(), R.string.sync_only_wifi_ssids_wifi_turn_on_wifi, Toast.LENGTH_LONG).show();
+        }
+
+        // On Android 8.1, ACCESS_COARSE_LOCATION is required, see issue #999
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
+            if (ContextCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+                if (getContext() instanceof Activity) {
+                    Activity activity = (Activity) getContext();
+                    ActivityCompat.requestPermissions(activity, new String[] { Manifest.permission.ACCESS_COARSE_LOCATION }, Constants.PERM_REQ_ACCESS_COARSE_LOCATION);
+                    this.getDialog().dismiss(); // wait for result
+                } else {
+                    Toast.makeText(getContext(), R.string.sync_only_wifi_ssids_need_to_grant_location_permission, Toast.LENGTH_LONG).show();
+                }
+            }
         }
     }
 

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -264,6 +264,11 @@ Please report any problems you encounter via Github.</string>
 
     <string name="sync_only_wifi_ssids_wifi_turn_on_wifi">Please turn on WiFi to select networks.</string>
 
+    <string name="sync_only_wifi_ssids_need_to_grant_location_permission">You need to grant LOCATION permission to use this feature.</string>
+
+    <string name="sync_only_wifi_ssids_location_permission_rejected_dialog_title">Permission required</string>
+    <string name="sync_only_wifi_ssids_location_permission_rejected_dialog_content">Starting with Android 8.1, location access is required to be able to read the WiFi\'s name. You can use this feature only if you grant this permission.</string>
+
     <string name="respect_battery_saving_title">Respect Android battery saving setting</string>
     <string name="respect_battery_saving_summary">Disable Syncthing if battery saving is active</string>
 

